pkg/api/transport.go

```diff
@@ -16,3 +16,12 @@ func NewTransport() *http.Transport {
 		ForceAttemptHTTP2:   true,
 	}
 }
+
+// GetOrCreateTransport gets the client's Transport if already exists, or initializes a new one.
+func GetOrCreateTransport(c *Client) *http.Transport {
```

Comment:
What if we pass a pointer to http.Transport to `NewTransport` and then do something like this:

```Go
func NewTransport(t *http.Transport) *http.Transport {
	if t != nil {
		return t.Clone()
	}

	return &http.Transport{
		Proxy:               nil,
		TLSHandshakeTimeout: 30 * time.Second,
		MaxIdleConns:        1,
		MaxIdleConnsPerHost: 1,
		MaxConnsPerHost:     1,
		ForceAttemptHTTP2:   true,
	}
}
```

Does that make sense @dron22 @alanhamlett?

Reply:
Sure, just combined the two functions into 1.